cassandra-five-zero-types/src/main/java/org/apache/cassandra/bridge/CassandraTypesImplementation.java

```diff
@@ -0,0 +1,352 @@
+import org.apache.cassandra.security.EncryptionContext;
+import org.apache.cassandra.spark.data.CassandraTypes;
+import org.apache.cassandra.spark.data.CqlField;
+import org.apache.cassandra.spark.data.complex.CqlCollection;
+import org.apache.cassandra.spark.data.complex.CqlFrozen;
+import org.apache.cassandra.spark.data.complex.CqlUdt;
+import org.apache.cassandra.spark.data.partitioner.Partitioner;
+import org.apache.cassandra.spark.data.types.Ascii;
+import org.apache.cassandra.spark.data.types.BigInt;
+import org.apache.cassandra.spark.data.types.Blob;
+import org.apache.cassandra.spark.data.types.Boolean;
+import org.apache.cassandra.spark.data.types.Counter;
+import org.apache.cassandra.spark.data.types.Date;
+import org.apache.cassandra.spark.data.types.Decimal;
+import org.apache.cassandra.spark.data.types.Double;
+import org.apache.cassandra.spark.data.types.Duration;
+import org.apache.cassandra.spark.data.types.Empty;
+import org.apache.cassandra.spark.data.types.Float;
+import org.apache.cassandra.spark.data.types.Inet;
+import org.apache.cassandra.spark.data.types.Int;
+import org.apache.cassandra.spark.data.types.SmallInt;
+import org.apache.cassandra.spark.data.types.Text;
+import org.apache.cassandra.spark.data.types.Time;
+import org.apache.cassandra.spark.data.types.TimeUUID;
+import org.apache.cassandra.spark.data.types.Timestamp;
+import org.apache.cassandra.spark.data.types.TinyInt;
+import org.apache.cassandra.spark.data.types.VarChar;
+import org.apache.cassandra.spark.data.types.VarInt;
+
+public class CassandraTypesImplementation extends CassandraTypes
+{
+    private static volatile boolean setup = false;
+
+    public static final CassandraTypesImplementation INSTANCE = new CassandraTypesImplementation();
+
+    public static synchronized void setup()
+    {
+        if (!CassandraTypesImplementation.setup)
+        {
+            // We never want to enable mbean registration in the Cassandra code we use so disable it here
+            System.setProperty("org.apache.cassandra.disable_mbean_registration", "true");
+            System.setProperty("cassandra.schema.force_load_local_keyspaces", "true"); // Cassandra 4.x vs 5.x
+            Config.setClientMode(true);
+            // When we create a TableStreamScanner, we will set the partitioner directly on the table metadata
+            // using the supplied IIndexStreamScanner.Partitioner. CFMetaData::compile requires a partitioner to
+            // be set in DatabaseDescriptor before we can do that though, so we set one here in preparation.
+            DatabaseDescriptor.setPartitionerUnsafe(Murmur3Partitioner.instance);
+            Config config = new Config();
+            config.memtable_flush_writers = 8;
+            config.memtable_heap_space = new DataStorageSpec.IntMebibytesBound(1024); // Cassandra 4.x vs 5.x
+            config.memtable_offheap_space = new DataStorageSpec.IntMebibytesBound(1024); // Cassandra 4.x vs 5.x
```

Comment:
I do not think they are relevant. The values could be smaller.

Reply:
Removed both. I am not sure how I ended up requiring them.